iroh/src/protocol.rs

```diff
@@ -311,6 +381,49 @@ impl Router {
         self.cancel_token.is_cancelled()
     }
 
+    /// Adds a protocol to the list of accepted protocols.
+    ///
+    /// Configures the router to accept the [`ProtocolHandler`] when receiving a connection
+    /// with this `alpn`.
+    ///
+    /// Once the function yields, new connections with this `alpn` will be handled.
+    ///
+    /// If a protocol handler was already registered for `alpn`, the previous handler will be shutdown.
+    ///
+    /// Returns `true` if
+    pub async fn accept(
+        &self,
+        alpn: impl AsRef<[u8]>,
+        handler: impl ProtocolHandler,
+    ) -> Result<AddProtocolOutcome, RouterError> {
+        let (reply, reply_rx) = oneshot::channel();
+        self.tx
+            .send(ToRouterTask::Accept {
+                alpn: alpn.as_ref().to_vec(),
+                handler: Arc::new(handler),
+                reply,
+            })
+            .await
+            .map_err(|_| RouterError::Closed {})?;
+        reply_rx.await.map_err(|_| RouterError::Closed {})
+    }
+
+    /// Stops accepting a protocol.
+    ///
+    /// Note that this has only an effect on new connections. Existing connections that were
+    /// accepted with `alpn` won't be closed when calling [`Router::stop_accepting`].
```

Comment:
Can you call out the differences in behaviour with calling `accept` where a previous handler was already handling this protocol?